rego/Commoncontrols.rego

```diff
@@ -1832,12 +1841,96 @@ tests contains {
 }
 if {
-    Events := {Event | some Event in ToggleServiceEvents; Event.OrgUnit == utils.TopLevelOU}
+    # Filter based on the service name of the event, otherwise all events are returned.
+    Events := {
+        Event | some Event in ToggleServiceEvents;
+        Event.OrgUnit == utils.TopLevelOU;
+        Event.ServiceName == "DISABLE_UNLISTED_SERVICES"
+    }
     count(Events) > 0
     Status := count(NonCompliantOUs16_1) == 0
 }
 #--
 
+#
+# Baseline GWS.COMMONCONTROLS.16.2v0.3
+#--
+
+NonCompliantOUs16_2 contains {
+    "Name": OU,
+    "Value": "Service status is ON"
+} if {
+    some OU in utils.OUsWithEvents
+    # Note that this setting requires the custom ToggleServiceEvents rule.
+    # Filter based on the service name of the event, otherwise all events are returned.
+    Events := {
+        Event | some Event in ToggleServiceEvents;
+        Event.OrgUnit == OU;
+        Event.ServiceName == "Early Access Apps"
+    }
+    # Ignore OUs without any events. We're already asserting that the
+    # top-level OU has at least one event; for all other OUs we assume
+    # they inherit from a parent OU if they have no events.
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    # For the Early Access Apps service:
+    # If service status is set to "ON for everyone", then "NewValue" == true (non-compliant state)
+    # else, "NewValue" == false (compliant state)
+    LastEvent.NewValue == "true"
+}
+
+NonCompliantGroups16_2 contains {
+    "Name": Group,
+    "Value": "Service status is ON"
+} if {
+    some Group in utils.GroupsWithEvents
+    # Note that this setting requires the custom ToggleServiceEvents rule.
+    Events := {
+        Event | some Event in ToggleServiceEvents;
+        Event.Group == Group;
+        Event.ServiceName == "Early Access Apps"
+    }
+    # Ignore groups without any events.
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    LastEvent.NewValue == "true"
+}
+
+tests contains {
+    "PolicyId": "GWS.COMMONCONTROLS.16.2v0.3",
+    "Criticality": "Should",
+    "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
+    "ActualValue": "No relevant event for the top-level OU in the current logs",
+    "RequirementMet": DefaultSafe,
+    "NoSuchEvent": true
+}
+if {
+    DefaultSafe := false
+    # Filter based on the service name of the event, otherwise all events are returned.
+    Events := {
+        Event | some Event in ToggleServiceEvents;
+        Event.OrgUnit == utils.TopLevelOU;
+        Event.ServiceName == "Early Access Apps"
+    }
+    count(Events) == 0
+}
+
+tests contains {
+    "PolicyId": "GWS.COMMONCONTROLS.16.2v0.3",
+    "Criticality": "Should",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs16_2, NonCompliantGroups16_2),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs16_2, "NonCompliantGroups": NonCompliantGroups16_2},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+} if {
+    Conditions := {
```

Comment:
This rule needs a `count(Events) > 0` statement somewhere to ensure that only one of the 16.2 `tests` blocks is matched. Right now if you run on a tenant where there are no events both blocks match and you get duplicate and contradictory output:
![image](https://github.com/user-attachments/assets/1f2687ae-1d48-4124-bc2c-80fde37cb935)

Reply:
Added the missing `count(Events) > 0` line